### DISCUSSION TITLE
Does it handle shutdown gracefully?

### DISCUSSION THREAD

**Discussion by zhangdong1015** [micronaut-projects/micronaut-core#7879] [General]
I'm trying to find any documentation on micronaut handling SIGTERM from Nomad gracefull termination. Does it handle it gracefully?

**Comment by alexanderankin** [comment]
Looks like graceful shutdown is implemented with the ApplicationShutdownEvent - https://github.com/micronaut-projects/micronaut-core/issues/2664 - https://docs.micronaut.io/latest/api/index.html?io/micronaut/runtime/event/ApplicationShutdownEvent.html
